### PR TITLE
Refactor/seeing stage

### PR DESCRIPTION
**Pull request by ergz**
implements the logic needed to get through the seeding stage of the model

### CODE REVIEW

File: R/model.R [hunk 1]
```diff
@@ -146,220 +160,368 @@ fall_run_model <- function(scenario = NULL, seeds = NULL){
...
                                       detour = 'sutter')
-
+        
         upper_sac_trib_rear <- rear(juveniles = upper_sac_trib_fish$inchannel,
                                     survival_rate = rearing_survival$inchannel[1:15, ],
                                     growth = growth_rates,
                                     floodplain_juveniles = upper_sac_trib_fish$floodplain,
                                     floodplain_survival_rate = rearing_survival$floodplain[1:15, ],
                                     floodplain_growth = growth_rates_floodplain,
                                     weeks_flooded = weeks_flooded[1:15, month, year])
```
Comment (ErinCain) [inline R/model.R]: For all the `rear()` functions do we need to use `juv_dynamics_year`? Year is used in the `weeks_flooded` input. `weeks_flooded[23, month, year]` I think that these years should be replaced with `juv_dynamics_year` in this rear and all the `rear()` below

File: R/model.R [hunk 2]
```diff
@@ -146,220 +160,368 @@ fall_run_model <- function(scenario = NULL, seeds = NULL){
...
+        
         migrants_at_golden_gate <- delta_fish$migrants_at_golden_gate
-
+        
         annual_migrants <- annual_migrants + migrants_at_golden_gate
+        
       } else {
+        
+        if (month == 11 & year > 1) {
+          yearlings <- yearling_growth(year, round(yearlings))
```
Comment (ErinCain) [inline R/model.R:172]: For `yearling_growth()` should we be using `yearling_dynamics_year()`? This one I am a little confused about but it looks like in the old model they do use the yr+1 in this case.

**Review by ErinCain — COMMENTED** [review]
Looks good. I had a couple comments about `juvenile_dynamics_year` vs `year` so we may want to spend a little more time making sure that we have that correct.

File: R/model.R (hunk 1 above)
Comment (ergz) [reply R/model.R]: good catch! yeah everything that references a year within the month iteration should be using the juv_dynamics_year

**Pull request merged**